src/store/reducers/swap.ts

```diff
@@ -66,20 +60,13 @@ const swapSlice = createSlice({
       state.swap.success = action.payload
       return state
     },
-    simulate(state, action: PayloadAction<Simulate>) {
-      state.swap.simulate = action.payload
-      return state
-    },
-    simulateSuccess(state, action: PayloadAction<boolean>) {
-      state.swap.simulate.success = action.payload
-      return state
-    },
     setPoolIndex(state, action: PayloadAction<number>) {
       state.swap.poolIndex = action.payload
       return state
     },
-    changePrice(state, action: PayloadAction<BN>) {
-      state.swap.price.v = action.payload
+    setPair(state, action: PayloadAction<{ tokenFrom: PublicKey; tokenTo: PublicKey }>) {
```

Comment:
we have tokenFrom and tokenTo parameters in onSwap callback, why can't it be set on swap request instead of adding next action?

Reply:
nevermind, i see it now